shark_turbine/kernel/wave/constraints.py

```diff
@@ -170,6 +170,13 @@ class WorkgroupConstraint(Constraint):
     tile_size: IndexExpr
     workgroup_dim: int
 
+    @property
+    def iterations(self) -> IndexExpr:
```

Comment:
since this is the total number of workgroups, should we just call it that? num_workgroups() ? (iterations is a little misleading since usually it is used in the context of a for-loop which in our case would be the tkw.reduction / TilingConstraint)

Reply:
I choose `iterations` specifically, because it's used interchangeably between `WorkgroupConstraint` and `TilingConstraint` in bounds check so I don't need to check for them separately. Maybe any ideas for better common name?

Reply:
How about counts? Specifies number of loop counts or workgroup counts?